synapse/rest/admin/users.py

```diff
@@ -195,6 +196,8 @@ async def on_PUT(self, request, user_id):
                 # add new threepids to user
                 current_time = self.hs.get_clock().time_msec()
                 for threepid in body["threepids"]:
+                    if threepid["medium"] == "email":
+                        threepid["address"] = canonicalise_email(threepid["address"])
```

Comment:
why is this necessary? doesn't `add_threepid` do it?

Reply:
also applies below

Reply:
yes, you're right